Add options to disable chat title and join/leave message sync

Manko10 opened the pull request:
The telegram sync bot is a bit too verbose. Things like chat titles and join/leave messages can add a lot of clutter and are not always wanted. This PR adds two options sync_join_messages sync_chat_titles which make this behavior configurable. They both default to true to not break existing functionality when they have not been set in the config file.

nylonee commented:
This will conflict with PR #642, I quite prefer this way of handling the sync names though, rather than removing them outright.
The code looks good though. I'll begin testing this on my bots

Manko10 commented:
I wonder why I never got a notification for your answer. I reapplied the changes to the new staging branch, so you should be able to merge now.
There was also a problem in PR #642 with the chat name sync. Messages from Telegram always had the username twice instead of the username and the chatname. That issue is fixed also.
I also added the "enable_sticker_sync" to the default configuration dict which is generated in _initialise().

Manko10 commented:
Jeez... Only one day and my few changes are conflicting again. I rebased once again and also fixed a regression in _on_hangouts_message() introduced by 86e3395af13f12ea9ba83521a193dfb3e66ba985 which threw an error when uploading photos from Hangouts to Telegram.
